src/pages/AdvancedSearch/ResultsBlock/ResultsBlock.tsx

```diff
@@ -50,22 +51,56 @@ const THEME = {
   base0F: DEFAULT_THEME_COLOUR,
 };
 
+const getTab = (key: string) => {
+  switch (key) {
+    case "applications":
+    case "offers":
+      return ModelTab.APPS;
+    case "machines":
+      return ModelTab.MACHINES;
+    case "relations":
+      return ModelTab.INTEGRATIONS;
+    default:
+      return;
+  }
+};
+
 const labelRenderer: LabelRenderer = (keyPath) => {
   const currentKey = keyPath[0];
+  // The last item in keyPath should always be the model UUID.
+  const lastItem = keyPath[keyPath.length - 1];
+  const modelUUID = typeof lastItem === "string" ? lastItem : null;
+  if (!modelUUID) {
+    // If this is somehow not a string then just display it.
+    return <>{currentKey}</>;
```

Comment:
I was wondering if maybe we could also add a test to cover this codepath?

Reply:
Done!